### ISSUE TITLE
SDK 48 Blank (Bare) template android  Couldn't open Android app with activity

### ISSUE THREAD

**Issue by cybercarrot** [expo/expo#21444]
### Minimal reproducible example

https://github.com/cybercarrot/bare-test.git

### Summary

when I create a bare-test project from Blank (Bare) template and 

> npx expo run:android

app BUILD SUCCESSFUL but CommandError: Couldn't open Android app with activity "com.baretest/.MainActivity" on device "Pixel_6_API_33".
The app might not be installed, try installing it with: npx expo run:android -d Pixel_6_API_33
Activity class {com.baretest/com.baretest.MainActivity} does not exist.
as pic below show. While iOS is ok. 
![WX20230228-231645@2x](https://user-images.githubusercontent.com/6837094/221897944-908a8f3a-a484-4d34-8d2b-6cfa825357bc.png)

### Environment

 expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.14.2/bin/yarn
      npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
      Watchman: 2023.02.13.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9514443
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.4 => 48.0.4 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.3 => 0.71.3 
    npmGlobalPackages:
      eas-cli: 3.7.2
    Expo Workflow: bare

**Comment by brentvatne** [comment]
it's possible that your emulator is out of disk space and perhaps we're not catching that error. try installing any another apk to it and i suspect you'll find that it won't work

**Comment by cybercarrot** [comment]
> it's possible that your emulator is out of disk space and perhaps we're not catching that error. try installing any another apk to it and i suspect you'll find that it won't work. can you run `DEBUG=* npx expo run:android` and share the full output?

hi [brentvatne](https://github.com/brentvatne), my emulator is new created and space is used as 2.6G,
![Screenshot_1677643253](https://user-images.githubusercontent.com/6837094/222041870-ef2c43c5-e5c7-4bbf-856f-8b7e435a53c7.png)


```
Resolved port: 8081, start dev server: true
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb devices -l +0ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 emu avd name +31ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb devices -l +51ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 emu avd name +29ms
  expo:android:resolveDevice Resolved default device (name: Pixel_6_API_33, pid: emulator-5554) +0ms
  expo:run:android Package name: com.baretest +0ms
› Building app...
  expo:start:platforms:android:gradle   /Users/hoo/Downloads/bare-test/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 +0ms
Configuration on demand is an incubating feature.

> Configure project :expo-application
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-constants
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-file-system
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-font
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-keep-awake
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-modules-core
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo-splash-screen
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :expo

Using expo modules
  - expo-application (5.1.1)
  - expo-constants (14.2.1)
  - expo-file-system (15.2.2)
  - expo-font (11.1.1)
  - expo-keep-awake (12.0.1)
  - expo-modules-core (1.2.3)
  - expo-splash-screen (0.18.1)

WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

The Kotlin Gradle plugin was loaded multiple times in different subprojects, which is not supported and may break the build.
This might happen in subprojects that apply the Kotlin plugins with the Gradle 'plugins { ... }' DSL if they specify explicit versions, even if the versions are equal.
Please add the Kotlin plugin to the common parent project or the root project, then remove the versions in the subprojects.
If the parent project does not need the plugin, add 'apply false' to the plugin line.
See: https://docs.gradle.org/current/userguide/plugins.html#sec:subprojects_plugins_dsl
The Kotlin plugin was loaded in the following projects: ':expo', ':expo-modules-core'

> Task :expo-constants:createDebugExpoConfig
Execution optimizations have been disabled for task ':expo-constants:createDebugExpoConfig' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:checkDebugAarMetadata' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:compileDebugAidl' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:compileDebugJavaWithJavac' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:compileDebugRenderscript' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:compileDebugShaders' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:createDebugCompatibleScreenManifests' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:extractDeepLinksDebug' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:generateDebugBuildConfig' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:generateDebugResValues' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:javaPreCompileDebug' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:mapDebugSourceSetPaths' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:mergeDebugNativeDebugMetadata' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:mergeDebugResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:mergeDebugShaders' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:processDebugMainManifest' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:processDebugManifest' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:processDebugManifestForPackage' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: '/Users/hoo/Downloads/bare-test/android'. Reason: Task ':expo-constants:createDebugExpoConfig' uses this output of task ':app:processDebugResources' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.5.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 1 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.

BUILD SUCCESSFUL in 12s
203 actionable tasks: 14 executed, 189 up-to-date
  expo:utils:scheme android application schemes: [] +0ms
  expo:utils:fileNotifier Observing /Users/hoo/Downloads/bare-test/babel.config.js +0ms
  expo:start:server:devServer Stopping dev server (bundler: metro) +0ms
  expo:start:server:devServer Stopped dev server (bundler: metro) +0ms

Expo Metro config:
- Version: 0.7.1
- Extensions: ts, tsx, js, jsx, json
- React Native: /Users/hoo/Downloads/bare-test/node_modules/react-native
- Babel config: /Users/hoo/Downloads/bare-test/babel.config.js
- Resolver Fields: react-native, browser, main
- Watch Folders:
- Node Module Paths:
- Exotic: false


Expo Metro config:
- Version: 0.7.1
- Extensions: ts, tsx, js, jsx, json
- React Native: /Users/hoo/Downloads/bare-test/node_modules/react-native
- Babel config: /Users/hoo/Downloads/bare-test/babel.config.js
- Resolver Fields: react-native, browser, main
- Watch Folders:
- Node Module Paths:
- Exotic: false

  expo:start:server:metro:router routerEntry /Users/hoo/Downloads/bare-test/node_modules/expo-router/entry /Users/hoo/Downloads/bare-test/app ../../app +0ms
  connect:dispatcher use / rawBodyMiddleware +0ms
  connect:dispatcher use / openURLMiddleware +0ms
  connect:dispatcher use / securityHeadersMiddleware +8ms
  connect:dispatcher use / compression +6ms
  connect:dispatcher use / nocache +0ms
  connect:dispatcher use /debugger-ui serveStatic +3ms
  connect:dispatcher use /launch-js-devtools devToolsMiddleware +0ms
  connect:dispatcher use / rawBodyMiddleware +0ms
  connect:dispatcher use / anonymous +0ms
  connect:dispatcher use /open-stack-frame handle +0ms
  connect:dispatcher use /open-url handle +0ms
  connect:dispatcher use /status statusPageMiddleware +0ms
  connect:dispatcher use /symbolicate rawBodyMiddleware +0ms
  connect:dispatcher use /systrace systraceProfileMiddleware +0ms
  connect:dispatcher use /reload anonymous +0ms
  connect:dispatcher use / errorHandler +2ms
  connect:dispatcher use / serveStatic +0ms
  connect:dispatcher use / remoteDevtoolsCorsMiddleware +1ms
  connect:dispatcher use / suppressRemoteDebuggingErrorMiddleware +0ms
  connect:dispatcher use / jsonParser +10ms
  connect:dispatcher use /logs anonymous +0ms
  connect:dispatcher use /inspector anonymous +1ms
  connect:dispatcher use / anonymous +0ms
Starting Metro Bundler
  connect:dispatcher use / processRequest +12ms
  connect:dispatcher use / handle +5ms
  connect:dispatcher use / bound processRequest +1ms
  connect:dispatcher use / middleware +19ms
  connect:dispatcher use / middleware +0ms
  connect:dispatcher use / middleware +0ms
  connect:dispatcher use / middleware +1ms
  expo:start:server:developmentSession Development session ping (runtime: native, url: http://localhost:8081) +0ms
  expo:api:fetch:base fetch: https://api.expo.dev/v2/development-sessions/notify-alive +0ms
  Metro:Watcher Beginning crawl with "watchman". +0ms
  Metro:Watcher Crawler "watchman" returned 209 added/modified, 0 removed, 1 clock(s). +156ms
  Metro:Watcher Using watcher: watchman +124ms
  Metro:WatchmanWatcher Received watch-project response: undefined +62ms
  Metro:WatchmanWatcher Received clock response: c:1676443936:18845:9:3022 +1ms
  Metro:WatchmanWatcher Received subscribe response: metro-file-map-60309--Users-hoo-Downloads-bare-test-e8df0a474cad4de68c8cffdd73518c8e +1ms
  expo:utils:fileNotifier Observing /Users/hoo/Downloads/bare-test/metro.config.js +1s
  expo:start:server:devServer Redirect page is disabled +1s
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █▄▀▀▄▄▀▀█▄█ ▄▄▄▄▄ █
█ █   █ ███▄█  ▀▀▄█ █   █ █
█ █▄▄▄█ ██▄▀▄▀███▀█ █▄▄▄█ █
█▄▄▄▄▄▄▄█ █ ▀▄▀ █ █▄▄▄▄▄▄▄█
█ ▄▀▄▄▀▄█▀ ▄▄▀▀█ ██▄▀▀▀▀▀▄█
█▀▀▀▀▀█▄▄▀▀  ▀█▄  ▄▄ ▀ ▀▀ █
█▀ ▄█▀ ▄▀▄▄█▄▄▀▄ ▀█▄▄▀██▀▄█
█ ▄▄   ▄▀▀██ ▄▄█ █▀  █▄ ▄ █
█▄████▄▄█ █ ▀▀▀▄█ ▄▄▄ █▄ ██
█ ▄▄▄▄▄ ██ █▄▀█▀▀ █▄█ ▄█▀ █
█ █   █ █ █▄█▄▄ ▄   ▄▄ █  █
█ █▄▄▄█ █▀▀▀█▄█ ▄▄▄█  █   █
█▄▄▄▄▄▄▄█▄▄███▄▄█▄▄▄██▄██▄█

› Metro waiting on http://localhost:8081
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press i │ open iOS simulator
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

  expo:start:platforms:android:adbServer [ADB] File output:
  expo:start:platforms:android:adbServer  arm64-v8a
 +17s
  expo:start:platforms:android:adb Property data: (device pid: emulator-5554, prop: ro.product.cpu.abilist, data: arm64-v8a
  expo:start:platforms:android:adb ) +0ms
  expo:run:android:resolveInstallApkName Supported ABIs: arm64-v8a, universal +0ms
  expo:run:android:resolveInstallApkName Searching for APK: /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug +0ms
  expo:run:android:resolveInstallApkName Checking for APK at: /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug/app-arm64-v8a-debug.apk +1ms
  expo:run:android:resolveInstallApkName Checking for APK at: /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug/app-universal-debug.apk +0ms
  expo:run:android:resolveInstallApkName Checking for fallback APK at: /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug/app-debug.apk +0ms
  expo:run:android Installing: /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug/app-debug.apk +17s
  connect:dispatcher handle  : /json +4s
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher suppressRemoteDebuggingErrorMiddleware  : /json +0ms
  connect:dispatcher remoteDevtoolsSecurityHeadersMiddleware  : /json +1ms
  connect:dispatcher compression  : /json +0ms
  connect:dispatcher nocache  : /json +0ms
  connect:dispatcher errorHandler  : /json +1ms
  connect:dispatcher serveStatic  : /json +1ms
  send stat "/Users/hoo/Downloads/bare-test/json" +0ms
  connect:dispatcher remoteDevtoolsCorsMiddleware  : /json +3ms
  connect:dispatcher jsonParser  : /json +0ms
  body-parser:json skip empty body +0ms
  connect:dispatcher <anonymous>  : /json +1ms
  connect:dispatcher processRequest  : /json +0ms
  Metro:Server Handling request: http://localhost:8081/json +0ms
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher bound processRequest  : /json +2ms
  compression no compression: size below threshold +0ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 install -r -d /Users/hoo/Downloads/bare-test/android/app/build/outputs/apk/debug/app-debug.apk +64ms
  connect:dispatcher handle  : /json +4s
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher suppressRemoteDebuggingErrorMiddleware  : /json +0ms
  connect:dispatcher remoteDevtoolsSecurityHeadersMiddleware  : /json +0ms
  connect:dispatcher compression  : /json +0ms
  connect:dispatcher nocache  : /json +1ms
  connect:dispatcher errorHandler  : /json +0ms
  connect:dispatcher serveStatic  : /json +0ms
  send stat "/Users/hoo/Downloads/bare-test/json" +4s
  connect:dispatcher remoteDevtoolsCorsMiddleware  : /json +0ms
  connect:dispatcher jsonParser  : /json +0ms
  body-parser:json skip empty body +4s
  connect:dispatcher <anonymous>  : /json +0ms
  connect:dispatcher processRequest  : /json +0ms
  Metro:Server Handling request: http://localhost:8081/json +4s
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher middleware  : /json +0ms
  connect:dispatcher bound processRequest  : /json +1ms
  compression no compression: size below threshold +4s
  expo:start:server:devServer Creating platform manager (platform: emulator, port: 8081) +10s
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb devices -l +7s
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 emu avd name +35ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 reverse tcp:8081 tcp:8081 +28ms
  expo:start:platforms:platformManager open (runtime: custom, platform: android, device: {
  expo:start:platforms:platformManager   pid: 'emulator-5554',
  expo:start:platforms:platformManager   name: 'Pixel_6_API_33',
  expo:start:platforms:platformManager   type: 'emulator',
  expo:start:platforms:platformManager   isAuthorized: true,
  expo:start:platforms:platformManager   isBooted: true
  expo:start:platforms:platformManager }, shouldPrompt: undefined) +0ms
  expo:start:platforms:platformManager open custom (applicationId: com.baretest) +1ms
  expo:start:platforms:platformManager Opening project in custom runtime: null -- { applicationId: 'com.baretest' } +0ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb devices -l +35ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 emu avd name +28ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell pm list packages com.baretest +24ms
› Opening com.baretest/.MainActivity on Pixel_6_API_33
  expo:start:platforms:android:activateWindow Activating window for device (pid: emulator-5554, type: emulator) +0ms
  expo:start:platforms:android:activateWindow lsof -i:5554 -P -t -sTCP:LISTEN +0ms
  expo:start:platforms:android:activateWindow Activate window for pid: 59260 +85ms
  expo:start:platforms:android:adbServer /Users/hoo/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell am start -a android.intent.action.RUN -f 0x20000000 -n com.baretest/.MainActivity +1s
CommandError: Couldn't open Android app with activity "com.baretest/.MainActivity" on device "Pixel_6_API_33".
The app might not be installed, try installing it with: npx expo run:android -d Pixel_6_API_33
Activity class {com.baretest/com.baretest.MainActivity} does not exist.

CommandError: Couldn't open Android app with activity "com.baretest/.MainActivity" on device "Pixel_6_API_33".
The app might not be installed, try installing it with: npx expo run:android -d Pixel_6_API_33
Activity class {com.baretest/com.baretest.MainActivity} does not exist.

    at openAsync (/Users/hoo/Downloads/bare-test/node_modules/@expo/cli/build/src/start/platforms/android/adb.js:112:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async AndroidDeviceManager.launchActivityAsync (/Users/hoo/Downloads/bare-test/node_modules/@expo/cli/build/src/start/platforms/android/AndroidDeviceManager.js:129:20)
    at async AndroidDeviceManager.openUrlAsync (/Users/hoo/Downloads/bare-test/node_modules/@expo/cli/build/src/start/platforms/android/AndroidDeviceManager.js:150:13)
    at async AndroidPlatformManager.openProjectInCustomRuntimeAsync (/Users/hoo/Downloads/bare-test/node_modules/@expo/cli/build/src/start/platforms/PlatformManager.js:97:9)
    at async runAndroidAsync (/Users/hoo/Downloads/bare-test/node_modules/@expo/cli/build/src/run/android/runAndroidAsync.js:44:5)
```

**Comment by brentvatne** [comment]
android emulators can be.. tricky. do you see the app that you build in the list of apps on the emulator? can you try installing any other arbitrary apk to see if that works?

**Comment by cybercarrot** [comment]
> android emulators can be.. tricky. do you see the app that you build in the list of apps on the emulator? can you try installing any other arbitrary apk to see if that works?

the app is installed on the emulator and open it show 
![Screenshot_1677650545](https://user-images.githubusercontent.com/6837094/222057958-f9ddebdd-1557-4cbc-aa9d-12b7de74e599.png)
so I run 
`npx expo run:ios`
to open metro server for it, it loaded Successfully!
![Screenshot_1677651121](https://user-images.githubusercontent.com/6837094/222059342-e9629e30-64bd-4e59-b7ad-4c4f7786c3d1.png)

**Comment by inmess** [comment]
I'm also facing an error with 
```
CommandError: Activity not started, unable to resolve Intent { act=android.intent.action.VIEW dat=exp+lp-connect://expo-development-client/... flg=0x10000000 }
```
![image](https://user-images.githubusercontent.com/48652450/222106832-a0452eec-3883-40e3-b1df-e4eb807961f6.png)

and when I tried to open it on a **real device**, it's quite weird...

![image](https://user-images.githubusercontent.com/48652450/222107380-0a54ee9c-cc98-4a0d-96e7-0e5d8cef2d19.png)

All occurred after I upgraded to SDK 48, and with prebuild. Does anyone have the same issue? (Nothing went wrong on IOS)

** Btw, not a blank project. I'll try to reproduce this within a new project later.

**Comment by alzo91** [comment]
Hello my dear friends.
Today I was testing expo SDK 48 and I noticed the same problem that was reported here.

It was created for the command line:
- [ ] npx create-expo-app --template bare-minimum
- [ ] :pensive: It didn't work today March 03

![image](https://user-images.githubusercontent.com/19477370/223567681-1b44b209-e496-4f42-b9b2-69a666f991c1.png)

In fact if I create a project using SDK 47 it's working correctly
- [x] :grinning: npx create-expo-app --template bare-minimum@~47

**Is there any solution for this problem?**

**Comment by iegik** [comment]
Seems to be you are changed the package name. So need to fix it:

android/app/src/main/AndroidManifest.xml

```diff
+++: package="real.package.name"
```

android/app/build.gradle

```diff
---:    namespace "com.baretest"
+++:    namespace "real.package.name"
```

android/app/build.gradle

```diff
---:    applicationId "com.baretest"
+++:    applicationId "real.package.name"
```

move *.java / *.kt files to correct path: `/real/package/name/`

**Comment by brentvatne** [comment]
i'm unable to reproduce this on a new project. it seems there are a number of distinct issues going on here - for example, as @iegik pointed out, some folks renamed their package name but didn't fully do it properly.

@lucy2329 - can you create a new issue and fill out the full template?

thanks!

**Comment by singh-sukhmanjit** [comment]
Was facing the same issue. I have given the fix here https://github.com/facebook/react-native/issues/36995

**Comment by marinadeotti** [comment]
**[Fix is here]( https://github.com/expo/expo/pull/22286/commits/6db41232ce9598115985a0355d8131376c6a00fa
)**.

Remove the lines:
```
 '-a',
 'android.intent.action.RUN',
```

from packages/@expo/cli/src/start/platforms/android/__tests__/adb-test.ts

**Comment by Parth0902** [comment]
> **[Fix is here](https://github.com/expo/expo/pull/22286/commits/6db41232ce9598115985a0355d8131376c6a00fa)**.
> 
> Remove the lines:
> 
> ```
>  '-a',
>  ' 'android.intent.action.RUN',',
> ```
> 
> from node_modules/@expo/cli/build/src/start/platforms/android/adb.js

there is  'android.intent.action.VIEW  in my aadb.js file .
I tried removing both the lines but still not working for me
![image](https://github.com/expo/expo/assets/102846775/913bb0b6-e2a4-450b-9e1e-98d9106144fd)
